### PR TITLE
Calculate inflation reward

### PR DESCRIPTION
**Pull request by karthikeyan-cere**

### CODE REVIEW

File: src/scenarios/scenarios_7.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    await this.fetchRewards(service);
+
+    this.tokenPerPoint = this.validatorReward / 10 ** CERE_DECIMAL / Number(this.totalEraRewardPoints);
+    await this.calculate();
+
+    this.logger.log(`Report ${JSON.stringify(this.entity)}`);
+  }
+
+  private async fetchValidatorAndNominator(service): Promise<any> {
+    const electedInfo = await service.fetchElectedInfo();
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_7.ts:47]: Validators and nominators should be taken from the previous era.

File: src/scenarios/scenarios_7.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  public async run() {
+    this.logger.log(`Starting scenario 7...`);
+
+    this.logger.log(`Initializing the blockchain`);
+    const service = new SubstrateService();
+    await service.initialize(cereTypes);
+
+    await this.fetchValidatorAndNominator(service);
+    await this.eraRewards(service, this.eraIndex);
```
[the remark this answers is not in the document]
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts]: Updated

File: src/scenarios/scenarios_7.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    await this.fetchRewards(service);
```
[the remark this answers is not in the document]
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts:38]: Updated

File: src/scenarios/scenarios_7.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts]: Updated

File: src/scenarios/scenarios_7.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    this.logger.log(`Initializing the blockchain`);
+    const service = new SubstrateService();
+    await service.initialize(cereTypes);
+
+    await this.fetchValidatorsAndNominatorsInfo(service,this.eraIndex);
+    await this.fetchEraRewards(service, this.eraIndex);
+    await this.fetchInflationRewards(service);
+
+    this.tokenPerPoint = this.validatorReward / 10 ** CERE_DECIMAL / Number(this.totalEraRewardPoints);
+    await this.calculateEarning();
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_7.ts:41]: Do you calculate one earning or earning**s** for validators and nominators? Please, rename

File: src/scenarios/scenarios_7.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+    this.logger.log(`Report ${JSON.stringify(this.entity)}`);
+  }
+
+  private async fetchValidatorsAndNominatorsInfo(service, eraIndex): Promise<any> {
+    const {validators} = await service.fetchValidatorsAndNominatorsInfo(eraIndex);
+    for (const validator in validators) {
+      const val = validator.toString();
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_7.ts:49]: Did not find a usage. Remove it

File: src/scenarios/scenarios_7.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      const vs = Number(validators[validator].own) / 10 ** CERE_DECIMAL;
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_7.ts:50]: What is `vs`. Rename it

**Review by AndreiNavoichyk — APPROVED** [review]

File: src/scenarios/scenarios_7.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    await this.eraRewards(service, this.eraIndex);
+    await this.fetchRewards(service);
+
+    this.tokenPerPoint = this.validatorReward / 10 ** CERE_DECIMAL / Number(this.totalEraRewardPoints);
+    await this.calculate();
+
+    this.logger.log(`Report ${JSON.stringify(this.entity)}`);
+  }
+
+  private async fetchValidatorAndNominator(service): Promise<any> {
```
[the remark this answers is not in the document]
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts]: Updated

File: src/scenarios/scenarios_7.ts (hunk 4 above)
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts]: Renamed

File: src/scenarios/scenarios_7.ts (hunk 5 above)
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts]: Removed

File: src/scenarios/scenarios_7.ts (hunk 6 above)
Comment (karthikeyan-cere) [reply src/scenarios/scenarios_7.ts]: Renamed

File: src/index.ts [hunk 8]
```diff
@@ -48,6 +49,12 @@ dotenv.config();
       await scenario.run();
       break;
     }
+    
+    case ScenarioEnum.SCENARIO_7: {
```
Comment (AndreiNavoichyk) [inline src/index.ts:53]: You should add it to the docker-compose

**Review by AndreiNavoichyk — APPROVED** [review]

File: src/index.ts (hunk 8 above)
Comment (AndreiNavoichyk) [reply src/index.ts:53]: Added

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**